src/components/JaegerIntegration/JaegerResults/JaegerItem.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import { sortBy } from 'lodash';
+import { Button, Card, CardBody, Label, Grid, GridItem } from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import { JaegerTrace, Span } from '../../../types/JaegerInfo';
+import { JaegerTraceTitle } from './JaegerTraceTitle';
+import { SpanDetail } from './SpanDetail';
+import { style } from 'typestyle';
+import moment from 'moment';
+import { formatDuration } from './transform';
+import { formatRelativeDate, cleanServiceSelector } from './transform';
+import { isErrorTag } from '../RouteHelper';
+import { KialiAppState } from '../../../store/Store';
+import { connect } from 'react-redux';
+import { PfColors } from '../../Pf/PfColors';
+
+const labelStyle = style({
+  margin: '5px'
+});
+
+interface JaegerScatterProps {
+  trace: JaegerTrace;
+  namespaceSelector: boolean;
+  namespace: string;
+  service: string;
+  maxTraceDuration: number;
+  jaegerURL: string;
+}
+
+interface JaegerScatterState {
+  spansSelected: Span[];
+  serviceSelected: string;
+}
+class JaegerItemC extends React.Component<JaegerScatterProps, JaegerScatterState> {
+  constructor(props: JaegerScatterProps) {
+    super(props);
+    this.state = { spansSelected: [], serviceSelected: '' };
+  }
+
+  getClassButtonSpan = (service: string) => {
+    if (this.state.serviceSelected === service) {
+      return 'primary';
+    } else {
+      const srv = this.props.namespaceSelector ? cleanServiceSelector(service, this.props.namespace) : service;
+      if (this.props.service === srv) {
+        return 'tertiary';
+      } else {
+        return 'secondary';
+      }
+    }
+  };
+
+  onClickService = (service: string) => {
+    this.setState({
+      serviceSelected: service,
+      spansSelected: this.props.trace.spans.filter(span => span.process.serviceName === service)
+    });
+  };
+
+  render() {
+    const { trace, maxTraceDuration } = this.props;
+    const { duration, services, spans, startTime } = trace;
+    const numSpans = spans.length;
+    const mDate = moment(startTime / 1000);
+    const fromNow = mDate.fromNow();
+    const timeStr = mDate.format('h:mm:ss a');
+    const numErredSpans = spans.filter(sp => sp.tags.some(isErrorTag)).length;
+    return (
+      <Card isCompact style={{ border: '1px solid #e6e6e6' }}>
+        <JaegerTraceTitle
+          trace={trace}
+          duration={duration}
+          durationPercent={(trace.duration / maxTraceDuration) * 100}
+          onClickLink={this.props.jaegerURL !== '' ? `${this.props.jaegerURL}/trace/${this.props.trace.traceID}` : ''}
+        />
+        <CardBody>
+          <Grid style={{ marginTop: '20px' }}>
+            <GridItem span={2}>
+              <Label isCompact>
```

Comment:
I find this compact labels hard to read in some standard resolution (100% zoom):
![image](https://user-images.githubusercontent.com/1662329/72426279-bb430400-3789-11ea-931d-37df177d731b.png)
I wonder if normal label looks better, but not a blocker, just a comment.